lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -1697,12 +1831,29 @@ func (d *zfs) MountVolume(vol Volume, op *operations.Operation) error {
 
 			revert.Add(func() { _ = d.setDatasetProperties(dataset, "volmode=none") })
 
-			// Wait half a second to give udev a chance to kick in.
-			time.Sleep(500 * time.Millisecond)
+			// Wait a second to give udev a chance to kick in.
+			time.Sleep(time.Second)
 
 			d.logger.Debug("Activated ZFS volume", logger.Ctx{"dev": dataset})
 		}
 
+		if vol.contentType != ContentTypeBlock && d.isBlockBacking(vol) && !filesystem.IsMountPoint(mountPath) {
+			err = vol.EnsureMountPath()
+			if err != nil {
+				return err
+			}
+
+			volPath, err := d.GetVolumeDiskPath(vol)
+			if err != nil {
+				return err
+			}
+
+			err = TryMount(volPath, mountPath, vol.ConfigBlockFilesystem(), 0, "")
```

Comment:
I would have expected to see something like:

```go
mountFlags, mountOptions := resolveMountOptions(vol.ConfigBlockMountOptions())
```

And then passing those to `TryMount` (like we do in the lvm) driver, as you did add `block.mount_options` support in the validation.

Reply:
I didn't actually know this existed. Thanks!